src/platform_impl/macos/view.rs

```diff
@@ -415,11 +417,8 @@ declare_class!(
             let base_y = (content_rect.origin.y + content_rect.size.height) as f64;
             let x = base_x + self.state.ime_position.x;
             let y = base_y - self.state.ime_position.y;
-            // This is not ideal: We _should_ return a different position based on
-            // the currently selected character (which varies depending on the type
-            // and size of the character), but in the current `winit` API there is
-            // no way to express this. Same goes for the `NSSize`.
```

Comment:
Nit: While I think this PR improves the situation, by "currently selected character" I think I actually meant the `range` parameter passed to this function.

Reply:
Hm, it's a bit weird that it's like that then, but I think our macOS backend doesn't even allow to move within the preedit string from what I remember, given that each time I test it the input is stuck on the last character, so I guess it's a separate issue, like usually the spot is set around the cursor in the editor and these character range is sort of redundant given that we manage it manually.

Reply:
I suspect the improper handling here is the reason the input is stuck at the end. But yeah, definitely a separate issue, and this PR does improve the situation.